pkg/plan/resource/kubeadm_init.go

```diff
@@ -233,10 +238,17 @@ func (ki *KubeadmInit) Undo(runner plan.Runner, current plan.State) error {
 	return buildKubeadmInitPlan(
 		remotePath,
 		strings.Join(ki.IgnorePreflightErrors, ","),
-		ki.UseIPTables, &ignored).Undo(runner, plan.EmptyState)
+		ki.UseIPTables, ki.KubernetesVersion, &ignored).Undo(
+		runner, plan.EmptyState)
 }
 
-func buildKubeadmInitPlan(path string, ignorePreflightErrors string, useIPTables bool, output *string) plan.Resource {
+func buildKubeadmInitPlan(path string, ignorePreflightErrors string, useIPTables bool, k8sVersion string, output *string) plan.Resource {
+	// Detect version for --upload-cert-flags
+	uploadCertsFlag := "--upload-certs"
+	if lt, err := version.LessThan(k8sVersion, "v1.15.0"); err == nil && lt {
+		uploadCertsFlag = "--experimental-upload-certs"
+	}
```

Comment:
I'm a bit confused: you changed `version` to `k8sversion` in many places, but here it seems to be the kubeadm version we are interested in?  Is it essential these versions are the same?

Reply:
according to the current behavior of `wksctl`, yes it is.

Reply:
Confirmed that it's Kubernetes version and not the version of Kubeadm.

Reply:
Sorry can you expand on what you're confirming?  `--upload-certs` is a flag on kubeadm.